org/mozilla/jss/ssl/SSLAlertDescription.java

```diff
@@ -4,57 +4,71 @@
+    HANDSHAKE_FAILURE               (40, SSLHandshakeException.class),
+    NO_CERTIFICATE                  (41, SSLPeerUnverifiedException.class), // SSL3 only, NOT TLS
+    BAD_CERTIFICATE                 (42, SSLPeerUnverifiedException.class),
+    UNSUPPORTED_CERTIFICATE         (43, SSLPeerUnverifiedException.class),
+    CERTIFICATE_REVOKED             (44, SSLPeerUnverifiedException.class),
+    CERTIFICATE_EXPIRED             (45, SSLPeerUnverifiedException.class),
+    CERTIFICATE_UNKNOWN             (46, SSLPeerUnverifiedException.class),
+    ILLEGAL_PARAMETER               (47, SSLProtocolException.class),
 
     // All alerts below are TLS only.
-    UNKNOWN_CA                      (48),
-    ACCESS_DENIED                   (49),
-    DECODE_ERROR                    (50),
-    DECRYPT_ERROR                   (51),
-    EXPORT_RESTRICTION              (60),
-    PROTOCOL_VERSION                (70),
-    INSUFFICIENT_SECURITY           (71),
-    INTERNAL_ERROR                  (80),
-    INAPPROPRIATE_FALLBACK          (86), // could also be sent for SSLv3
-    USER_CANCELED                   (90),
-    NO_RENEGOTIATION                (100),
+    UNKNOWN_CA                      (48, SSLPeerUnverifiedException.class),
+    ACCESS_DENIED                   (49, SSLHandshakeException.class),
+    DECODE_ERROR                    (50, SSLProtocolException.class),
+    DECRYPT_ERROR                   (51, SSLProtocolException.class),
+    EXPORT_RESTRICTION              (60, SSLHandshakeException.class),
+    PROTOCOL_VERSION                (70, SSLHandshakeException.class),
+    INSUFFICIENT_SECURITY           (71, SSLHandshakeException.class),
+    INTERNAL_ERROR                  (80, SSLProtocolException.class),
+    INAPPROPRIATE_FALLBACK          (86, SSLProtocolException.class), // could also be sent for SSLv3
+    USER_CANCELED                   (90, SSLProtocolException.class),
+    NO_RENEGOTIATION                (100, SSLHandshakeException.class),
 
     // Alerts for client hello extensions
-    MISSING_EXTENSION               (109),
-    UNSUPPORTED_EXTENSION           (110),
-    CERTIFICATE_UNOBTAINABLE        (111),
-    UNRECOGNIZED_NAME               (112),
-    BAD_CERTIFICATE_STATUS_RESPONSE (113),
-    BAD_CERTIFICATE_HASH_VALUE      (114),
-    NO_APPLICATION_PROTOCOL         (120);
+    MISSING_EXTENSION               (109, SSLHandshakeException.class),
+    UNSUPPORTED_EXTENSION           (110, SSLHandshakeException.class),
+    CERTIFICATE_UNOBTAINABLE        (111, SSLPeerUnverifiedException.class),
+    UNRECOGNIZED_NAME               (112, SSLHandshakeException.class),
+    BAD_CERTIFICATE_STATUS_RESPONSE (113, SSLPeerUnverifiedException.class),
+    BAD_CERTIFICATE_HASH_VALUE      (114, SSLPeerUnverifiedException.class),
+    NO_APPLICATION_PROTOCOL         (120, SSLHandshakeException.class);
 
     private int id;
+    private Class exception;
```

Comment:
Does this generate a warning in Eclipse? Usually a `Class` has to be specified as `Class<?>`.

Reply:
@edewata I've updated this to be `Class<? extends SSLException>` since that's what actually happens... :-)